datafusion/physical-expr/src/intervals/interval_aritmetic.rs

```diff
@@ -194,6 +194,18 @@ impl Display for IntervalBound {
 /// called *unbounded* endpoint and represented using a `NULL` and written using
 /// `∞`.
 ///
+/// # Boolean Handling
+///
+/// Boolean values require special handling.
+///
+/// Given there are only two  boolean values, and they are ordered such that
+/// `false` is less than `true`, there are only three possible valid intervals
+/// for a boolean `[false, false]`, `[false, true]` or `[true, true]`, all with
```

Comment:
The core of my confusion is that boolean `Intervals` NEVER have open bounds -- and if you try to make one with open bounds, `Interval::new` will remap them